libcfnet/classic.c

```diff
@@ -100,12 +101,17 @@ int RecvSocketStream(int sd, char buffer[CF_BUFSIZE], int toget)
                 return -1;
             }
         }
-        else if (got == 0)                /* peer has closed the connection */
+        else if (got == 0)          /* peer has closed the connection early */
         {
-            break;
+            Log(LOG_LEVEL_ERR,
+                "Peer closed connection early (received=%dB, expecting=%dB)",
+                already, toget);
+            buffer[already] = '\0';
+            /* TODO Why return 0? Early close is an error so -1 is better. */
+            return 0;
         }
     }
-    assert(already <= toget);
+    assert(already == toget);
```

Comment:
It is a programming error if this is the case, that's why I'm using assert. If it was dependant on any external factor, it would be vital to be checked in prod code as well, but it's not.